packages/auto-approve/src/auto-approve.ts

```diff
@@ -0,0 +1,226 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file handles the logic to manage incoming pull-requests
+
+// eslint-disable-next-line node/no-extraneous-import
+import {Probot, Context, ProbotOctokit} from 'probot';
+import {logger} from 'gcf-utils';
+import {ValidPr, checkPRAgainstConfig} from './check-pr';
+import {getChangedFiles, getBlobFromPRFiles} from './get-PR-info';
+import {validateYaml, validateSchema, checkCodeOwners} from './check-config.js';
+
+interface Configuration {
+  rules: ValidPr[];
+}
+const CONFIGURATION_FILE_PATH = 'auto-approve.yml';
+
+/**
+ * Takes in the auto-approve.yml file and (if it exists) the CODEOWNERS file
+ * and checks both files to ensure they match schema, format, and appropriate CODEOWNERS
+ *
+ * @param owner owner of the repo of the incoming PR
+ * @param repo string, the name of the repo of the incoming PR
+ * @param codeOwnersFile string, the CODEOWNERS file in .github/CODEOWNERS for the repo of the incoming PR
+ * @param octokit the octokit instance
+ * @param headSha the sha upon which to check whether the config and the CODEOWNERS file are configured correctly
+ * @returns true if the status check passed, false otherwise
+ */
+export async function logicForConfigCheck(
```

Comment:
I'd probably use a more specific verb here, like `validateAutoApproveFile`, I'd probably also break it into two methods:

```
validateAutoApproveFile()
validateCodeOwnersFile()
```

Reply:
Not sure if you've taken a look at src/check-config.ts, which actually does the heavy lifting and is broken up into three separate methods (`validateYaml`, `validateSchema`, and `checkCodeOwners`). This is just a wrapper function that makes sure that all three are true before submitting a passing check, else it submits a failing one. Beyond that, do you still feel that we should break up this function?

Reply:
Not a blocker; I think what jumped out at me is that `logic` is a little vague, it could mean *anything*.

I think the term `handle` might resonate with me more, because I'm used to the term handler being used to refer to something that performs the logic for a route -- I realize now that I type this that handler is pretty much as vague as "logic for" 😛